transmogrifier/sources/xml/datacite.py

```diff
@@ -339,6 +242,189 @@ def _get_contributors(cls, source_record: Tag) -> Iterator[timdex.Contributor]:
+                "Datacite record %s missing required Datacite field publicationYear",
+                cls.get_source_record_id(source_record),
+            )
+
+    @classmethod
+    def _get_dates_by_date_element(cls, source_record: Tag) -> Iterator[timdex.Date]:
+        for date_element in source_record.metadata.find_all("date"):
+            date_object = timdex.Date()
+            if date_value := date_element.string:
+                date_value = str(date_value)
+                if "/" in date_value:
+                    date_object = cls._parse_date_range(
+                        date_object, date_value, cls.get_source_record_id(source_record)
+                    )
+                else:
+                    date_object.value = (
+                        date_value.strip()
+                        if validate_date(
+                            date_value,
+                            cls.get_source_record_id(source_record),
+                        )
+                        else None
+                    )
+            date_object.note = date_element.get("dateInformation") or None
+            if any([date_object.note, date_object.range, date_object.value]):
+                date_object.kind = date_element.get("dateType") or None
+                yield date_object
+
+    @classmethod
+    def _parse_date_range(
+        cls, date_object: timdex.Date, date_value: str, source_record_id: str
+    ) -> timdex.Date:
+        split = date_value.index("/")
+        gte_date = date_value[:split].strip()
+        lte_date = date_value[split + 1 :].strip()
+        if validate_date_range(
+            gte_date,
+            lte_date,
+            source_record_id,
+        ):
+            date_object.range = timdex.DateRange(
+                gte=gte_date,
+                lte=lte_date,
+            )
+        return date_object
+
+    @classmethod
+    def get_edition(cls, source_record: Tag) -> str | None:
+        if edition := source_record.metadata.find("version", string=True):
+            return str(edition.string)
+        return None
+
+    @classmethod
+    def get_file_formats(cls, source_record: Tag) -> list[str] | None:
+        return [
+            str(file_format.string)
+            for file_format in source_record.metadata.find_all("format", string=True)
+        ] or None
+
+    @classmethod
+    def get_format(cls) -> str:
+        return "electronic resource"
+
+    @classmethod
+    def get_funding_information(cls, source_record: Tag) -> list[timdex.Funder] | None:
+        funding_information = []
+        for funding_reference in source_record.metadata.find_all("fundingReference"):
+            funder = timdex.Funder()
+            if funder_name := funding_reference.find("funderName", string=True):
+                funder.funder_name = str(funder_name.string)
+            if award_number := funding_reference.find("awardNumber"):
+                funder.award_number = award_number.string or None
```

Comment:
I should have made a note about this but forgot, `str(award_number.string)` returns `"None"` if there is no `.string`. However, we can't do `funding_reference.find("awardNumber", string=True)` because we're also interested in the `awardURI` attribute even if there isn't `award_number.string`. Not sure if there is a better way to handle this but decided to make a one-off exception that rule, happy to update if anyone has suggestions!